sync/syncer.go

```diff
@@ -398,12 +404,15 @@ func (s *Syncer) handleNotSynced(currentSyncLayer types.LayerID) {
 
 		lyr, err := s.getLayerFromNeighbors(currentSyncLayer)
 		if err != nil {
-			s.Info("could not get layer %v from neighbors: %v", currentSyncLayer, err)
+			s.With().Info("could not get layer from neighbors", log.LayerId(currentSyncLayer.Uint64()), log.Err(err))
 			return
 		}
 
 		if len(lyr.Blocks()) == 0 {
-			s.SetZeroBlockLayer(currentSyncLayer)
+			if err := s.SetZeroBlockLayer(currentSyncLayer); err != nil {
+				s.With().Info("could not get layer from neighbors ", log.LayerId(currentSyncLayer.Uint64()), log.Err(err))
```

Comment:
I don't get this error. We get here if we _have_ received an answer from our neighbours. The error should be related to setting the layer empty, doesn't it?

Reply:
this can be caused by peers timing out, ill change the txt to something clearer

Reply:
if there was a timeout error we won't get here, would we? we will break by the previous return

Reply:
timeout is not considered error yet, and we will get here because the time out happend in a previous call to synchronise

Reply:
the time out was in handleNotSynced

Reply:
this thread in under handleNotSynced... 

ok, so why timeouts aren't an error (for consistency: the timeout log print is error)?
basically we may set a layer zero just because we got timeouts from our peers.

Reply:
implementation wise its hard to do now, initially there was no need, all of this is because this happens async kinda, lets fix that in a different pr i dont think it affects the flow

Reply:
sorry i confused you timeouts are an error its just an error of the previos call to syncronise()